packages/web-components/src/components/rux-container/rux-container.tsx

```diff
@@ -0,0 +1,109 @@
+import { State, Component, Host, h, Element } from '@stencil/core'
+import { hasSlot } from '../../utils/utils'
+
+/**
+ * @slot (default) - The container's content
+ * @slot header - The container's header
+ * @slot tab-bar - The container's tab bar
+ * @slot toolbar - The container's toolbar
+ * @slot footer - The container's footer
+ * @part header - The container's outside header element
+ * @part tab-bar - The container's outside tab bar element
+ * @part toolbar - The container's outside toolbar element
+ * @part body - The container's outside body element
+ * @part footer - The container's outside footer element
+ *
+ */
+@Component({
+    tag: 'rux-container',
+    styleUrl: 'rux-container.scss',
+    shadow: true,
+})
+export class RuxContainer {
+    @Element() el!: HTMLRuxContainerElement
+
+    @State() activeSlots = {
+        header: false,
+        'tab-bar': false,
+        toolbar: false,
+        footer: false,
+    }
+
+    connectedCallback() {
+        this._handleSlotChange = this._handleSlotChange.bind(this)
+    }
+
+    private _handleSlotChange(
+        slotName: 'header' | 'tab-bar' | 'toolbar' | 'footer'
+    ) {
+        const show = hasSlot(this.el, slotName)
+        this.activeSlots = { ...this.activeSlots, [slotName]: show }
```

Comment:
idk this is kind of too clever and not as readable as just creating separate hasXXSlot states. But we should we removing a lot of them soon